From abe024653bd24f289d1e14a24a8e8468f01d4012 Mon Sep 17 00:00:00 2001
From: David Mabe <dmmabe@ncsu.edu>
Date: Sun, 29 Mar 2015 18:46:59 -0400
Subject: [PATCH] Disable iossim build to fix linker errors.

Patch from Robin Sommer <robin@icir.org> re-rolled to LLVM 3.6.

When attempting to build the address sanitizer for the iOS simulator,
we run into an architecture mismatch between dependent shared
libraries, such as libc++.dylib.

Disabling the iOS simulator build fixes the issue.
---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f8f5263..fbc84e8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -304,7 +304,9 @@ if(APPLE)
          MACOSX_VERSION_MIN_FLAG "${CMAKE_CXX_FLAGS}")
   set(SANITIZER_COMMON_SUPPORTED_DARWIN_OS osx)
   if (IOSSIM_SDK_DIR AND NOT MACOSX_VERSION_MIN_FLAG)
-    list(APPEND SANITIZER_COMMON_SUPPORTED_DARWIN_OS iossim)
+    # Disable iOS Simulator build because it clashes with the architecture of
+    # dependent libraries, such libc++abi and libc++.
+    # list(APPEND SANITIZER_COMMON_SUPPORTED_DARWIN_OS iossim)
   endif()
 
   set(SANITIZER_MIN_OSX_VERSION 10.7)
-- 
1.9.5 (Apple Git-50.3)

